pkg/state/leases.go

```diff
@@ -37,38 +41,19 @@ func (lr *leaseRecordForStateHashes) less(other stateComponent) bool {
 }
 
 type leasing struct {
-	isActive    bool
-	leaseAmount uint64
-	recipient   proto.Address
-	sender      proto.Address
-}
-
-type leasingRecord struct {
-	leasing
-}
-
-func (l *leasingRecord) marshalBinary() ([]byte, error) {
-	res := make([]byte, leasingRecordSize)
-	proto.PutBool(res[0:1], l.isActive)
-	binary.BigEndian.PutUint64(res[1:9], l.leaseAmount)
-	copy(res[9:9+proto.AddressSize], l.recipient[:])
-	copy(res[9+proto.AddressSize:9+proto.AddressSize*2], l.sender[:])
-	return res, nil
+	Sender              proto.Address  `cbor:"0,keyasint"`
+	Recipient           proto.Address  `cbor:"1,keyasint"`
+	Amount              uint64         `cbor:"2,keyasint"`
+	Height              uint64         `cbor:"3,keyasint"`
+	Status              LeaseStatus    `cbor:"4,keyasint"`
+	OriginTransactionID *crypto.Digest `cbor:"5,keyasint,omitempty"`
+	RecipientAlias      *proto.Alias   `cbor:"6,keyasint,omitempty"`
+	CancelHeight        uint64         `cbor:"7,keyasint,omitempty"`
+	CancelTransactionID *crypto.Digest `cbor:"8,keyasint,omitempty"`
 }
 
-func (l *leasingRecord) unmarshalBinary(data []byte) error {
-	if len(data) != leasingRecordSize {
-		return errInvalidDataSize
-	}
-	var err error
-	l.isActive, err = proto.Bool(data[0:1])
-	if err != nil {
-		return err
-	}
-	l.leaseAmount = binary.BigEndian.Uint64(data[1:9])
-	copy(l.recipient[:], data[9:9+proto.AddressSize])
-	copy(l.sender[:], data[9+proto.AddressSize:9+proto.AddressSize*2])
-	return nil
+func (l leasing) isActive() bool {
+	return l.Status == LeaseActive
```

Comment:
this construction is weird. Shouldn't it be ``` return l.Status``` ?

Reply:
Nope, one status is active, the other two is inactive. For now we only want to know is it active or not. May be we will have to remove third status and move back to simple bool value. Don't know.